### PR TITLE
describe command

### PR DESCRIPTION
**Pull request by sheldon**
Hi

I was struggling to find an easy way to browse all the available packages and find out what they do, so I added an info command to try achieve this.

It pulls the descriptions from the repos on github, so is a little slow. Potentially we could add this info into the local db to make it super quick, but I didn't want to change formats without discussing first.

What do you think?

### CODE REVIEW

File: pkg/omf/cli/omf.info.fish [hunk 1]
```diff
@@ -0,0 +1,16 @@
+function omf.info -a name
+  if test (count $argv) -eq 0
+    for package in (omf.list_db_packages)
+      echo $package - (omf.info $package)
+    end
+  else
+    set url (cat $OMF_PATH/db/pkg/$name)
+    set repo (basename (dirname $url))/(basename $url)
+    curl -s https://api.github.com/repos/$repo | ruby -rjson -e 'j = JSON.parse(ARGF.read); puts j["description"]'
+    or begin
```
Comment (bucaran) [inline pkg/omf/cli/omf.info.fish]: Let's go with...
```fish
if not curl...
...
end
```

File: pkg/omf/cli/omf.info.fish [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+      echo (omf::err)"$local_name is not a valid pkg."(omf::off) 1^&2
+      return $OMF_INVALID_ARG
+    end
+  end
+  return 0
```
Comment (bucaran) [inline pkg/omf/cli/omf.info.fish]: Is this return really necessary?

File: pkg/omf/omf.fish [hunk 3]
```diff
@@ -72,6 +72,13 @@ function omf -d "Oh My Fish"
     case "l" "li" "lis" "lst" "list"
       omf.list_local_packages | column
 
+    case "info"
```
Comment (bucaran) [inline pkg/omf/omf.fish]: The `i` in `info` collides with `install`. Let's find another name?

**Comment by bucaran** [comment]
LGTM.

> It pulls the descriptions from the repos on github, so is a little slow. Potentially we could add this info into the local db to make it super quick, but I didn't want to change formats without discussing first.

I decided against this early on because it would increase complexity significantly to parse the registry. It's a simple text file.

Another thing is, we should remember that OMF is working kind of like a package manager for shell utilities / configuration, but it shouldn't really be a package manager. That's what you use `brew` or whatever it is you use for.

File: pkg/omf/cli/omf.info.fish (hunk 2 above)
Comment (sheldon) [inline pkg/omf/cli/omf.info.fish]: good point, dropped it out

File: pkg/omf/omf.fish (hunk 3 above)
Comment (derekstavis) [inline pkg/omf/omf.fish]: We could use `d`, `de`, `describe`.

Comment (bucaran) [inline pkg/omf/omf.fish]: Options

+ `d` for describe
+ `e` for explain
+ `a` for about

Or keep `info` and use `f`, and `inf` for the aliases.

**Comment by bpinto** [comment]
I'll test it later but I like the idea!

You need to add this command to help.

File: pkg/omf/cli/omf.info.fish (hunk 1 above)
Comment (sheldon) [inline pkg/omf/cli/omf.info.fish]: nice spot, the error handling was all wrong, I've updated it now to properly check for the package existing

File: pkg/omf/omf.fish (hunk 3 above)
Comment (sheldon) [inline pkg/omf/omf.fish]: sure, how about describe

**Comment by sheldon** [comment]
@bucaran good point, I think the slow response isn't too bad anyway, and you can ask for 1 exact package directly for a quick output

@bpinto added in the help now

**Comment by bucaran** [comment]
@sheldon And to completions! :smile: 

@bpinto The same goes for `cd`.

**Comment by sheldon** [comment]
@bucaran @bpinto added in the completions, and cd too :smile:

**Comment by bucaran** [comment]
@bpinto Maybe `d` is not a good idea? Because it can be interpreted as `delete`? 

@sheldon Sorry about the indecisiveness.

**Comment by sheldon** [comment]
no problem, makes sense the clash

**Comment by sheldon** [comment]
the other 2 suggestions of about or explain both work without the clash, any preferences?

**Comment by bpinto** [comment]
Info or describe are the best. Regarding these short commands, I am still going evaluating it in my head. I honestly don't like and don't use any myself.

**Comment by bpinto** [comment]
Thanks for the cd autocomplete but there is one more missing for both commands, the one that will complete the package/theme name.

**Comment by sheldon** [comment]
added in showing the readme of the repo when describing just 1 package

**Comment by sheldon** [comment]
@bpinto sure, I've added in the package name completions too

**Comment by bpinto** [comment]
Full readme is unnecessary?
```
⋊> omf on sheldon-info-command ◦ omf describe ansible                                                                                                                     15:00:53
:black_circle: Ansible development package for Wahoo

[![](https://img.shields.io/badge/Wahoo-Package-00b0ff.svg?style=flat-square)][Wahoo]
![](https://img.shields.io/badge/License-MIT-707070.svg?style=flat-square)

# :black_circle: [_ansible_](https://github.com/ansible/ansible)
> Ansible is a powerful automation tool that you can learn quickly.

Ansible development for [Wahoo][Wahoo] / [fish](fishshell.com).

## Install
> _Note_: You need to have `ansible` source code handy and set `ANSIBLE_HOME` prior to loading Wahoo.
```

**Comment by bpinto** [comment]
By the way, I like this very much! :D

**Comment by sheldon** [comment]
sure, the full readme is a little overkill, I got carried away! :)

**Comment by bpinto** [comment]
@sheldon 
Rebase so we can merge?
Squashing your commits would also be interesting.

**Comment by sheldon** [comment]
sure, all in 1 commit now

**Comment by derekstavis** [comment]
[LGTM](http://lgtm.in/i/owaBPNIcq)

**Comment by bpinto** [comment]
Thanks @sheldon !

**Comment by derekstavis** [comment]
Now you was faster, @bpinto

**Pull request merged**